src/cogent3/core/new_alignment.py

```diff
@@ -4111,6 +4359,361 @@ def get_features(
+
+        drawer = Drawable(title=title, traces=all_traces, width=width, height=height)
+        drawer.layout.update(xaxis=xaxis, yaxis=yaxis)
+        return drawer
+
+    def seqlogo(
+        self,
+        width: float = 700,
+        height: float = 100,
+        wrap: OptInt = None,
+        vspace: float = 0.005,
+        colours: dict = None,
+    ):
+        """returns Drawable sequence logo using mutual information
+
+        Parameters
+        ----------
+        width, height
+            plot dimensions in pixels
+        wrap
+            number of alignment columns per row
+        vspace
+            vertical separation between rows, as a proportion of total plot
+        colours
+            mapping of characters to colours. If note provided, defaults to
+            custom for everything ecept protein, which uses protein moltype
+            colours.
+
+        Notes
+        -----
+        Computes MI based on log2 and includes the gap state, so the maximum
+        possible value is -log2(1/num_states)
+        """
+        assert 0 <= vspace <= 1, "vertical space must be in range 0, 1"
+        freqs = self.counts_per_pos(allow_gap=True).to_freq_array()
+        if colours is None and "protein" in self.moltype.label:
+            colours = self.moltype._colors
+
+        return freqs.logo(
+            width=width, height=height, wrap=wrap, vspace=vspace, colours=colours
+        )
+
+    @c3warn.deprecated_args(
+        "2024.12", reason="consistency", old_new=[("method", "stat")]
+    )
+    def coevolution(
+        self,
+        stat: str = "nmi",
+        segments: list[tuple[int, int]] = None,
+        drawable: OptStr = None,
+        show_progress: bool = False,
+        parallel: bool = False,
+        par_kw: OptDict = None,
+    ):
+        """performs pairwise coevolution measurement
+
+        Parameters
+        ----------
+        stat
+            coevolution metric, defaults to 'nmi' (Normalized Mutual
+            Information). Valid choices are 'rmi' (Resampled Mutual Information)
+            and 'mi', mutual information.
+        segments
+            coordinates of the form [(start, end), ...] where all possible
+            pairs of alignment positions within and between segments are
+            examined.
+        drawable
+            Result object is capable of plotting data specified type. str value
+            must be one of plot type 'box', 'heatmap', 'violin'.
+        show_progress
+            shows a progress bar.
+        parallel
+            run in parallel, according to arguments in par_kwargs.
+        par_kw
+            dict of values for configuring parallel execution.
+
+        Returns
+        -------
+        DictArray of results with lower-triangular values. Upper triangular
+        elements and estimates that could not be computed for numerical reasons
+        are set as nan
+        """
+        from cogent3.draw.drawable import AnnotatedDrawable, Drawable
```

Comment:
add a comment here about redesign. These graphical representations of matrices should be separate functions in the drawing submodule somewhere